### ISSUE TITLE
services/horizon: change configuration of SKIP_TXMETA in puppet to true

### ISSUE THREAD

**Issue by mollykarcher** [stellar/go#5452]
### What problem does your feature solve?

We announced back in January that we would be sunsetting txmeta from Horizon, in anticipation of a potential for a large increase in data size following the launch of Soroban. The ecosystem alternative should they need access to txmeta is to get it from RPC's `getTransactions` endpoint, and this has been available since [21.3.0](https://github.com/stellar/soroban-rpc/releases/tag/v21.3.0) (~mid June).  

### What would you like to see?

Change the default for the SKIP_TXMETA flag to true, so that txmeta is no longer returned by Horizon by default.

### What alternatives are there?

- Leave the default as `false` and simply set the config to `true` on SDF's Horizon deployment. Even though this isn't what we had originally planned, I think that given we haven't seen the explosion in data size due to Soroban that was anticipated, this might be the better option to go with for now, given it's less disruptive overall. And it still de-risks us from our deployment's perspective, since we don't need to store/service it on ours. Open to thoughts here.

**Comment by sreuland** [comment]
Any preference on rollout of these changes for the 3 horizon deployments, i.e. [futurenet](https://horizon-futurenet.stellar.org) first, [testnet](https://horizon-testnet.stellar.org) second, [pubnet](https://horizon.stellar.org) third.

With the rollout commencing right after or before the P22 is voted?

**Comment by JakeUrban** [comment]
I agree @mollykarcher, I don't see a strong reason to delay the update of the `SKIP_TXMETA` flag on our testnet deployment to the p22 vote. Bundling it with the rollout of our blog post & upgrade guide makes sense. @briwylde08 we'll want to update our guide to note that txmeta is being removed from SDF horizon instances.

**Comment by sreuland** [comment]
The horizon testnet update for enabling SKIP_TXMETA is staged - https://github.com/stellar/puppet-v4/pull/4119

@johncanneto , let me know what preconditions need to be met before we apply to [hosted horizon testnet](https://horizon-testnet.stellar.org), if it's a specific RC release for core or rpc, or other p22 preparations, thanks!

**Comment by sreuland** [comment]
SKIP_TXMETA=true has been deployed onto horizon testnet - https://github.com/stellar/puppet-v4/pull/4119

will do horizon pubnet later when appropriate per the protocol 22 upgrade schedule.

**Comment by sreuland** [comment]
draft pr's are staged for updating prd horizon cluster configs for skip tx-meta, just need to merge these when ready:
https://github.com/stellar/puppet-v4/pull/4168
https://github.com/stellar/puppet-v4/pull/4169

**Comment by tamirms** [comment]
We will merge the PRs on Dec 5, 2024

**Comment by sreuland** [comment]
this has been fully merged to prod deployment, closing.
